objectstore.go

```diff
@@ -13,49 +15,40 @@ import (
 )
 
 const (
-	accessGrant = "accessGrant"
+	accessGrant        = "accessGrant"
+	linksharingBaseURL = "https://link.tardigradeshare.io"
 )
 
 type ObjectStore struct {
-	log     logrus.FieldLogger
-	project *uplink.Project
+	log    logrus.FieldLogger
+	access *uplink.Access
 }
 
 func newObjectStore(logger logrus.FieldLogger) *ObjectStore {
 	return &ObjectStore{log: logger}
 }
 
-func setupUplink(ctx context.Context, storjAccessGrant string) (*uplink.Project, error) {
-	access, err := uplink.ParseAccess(storjAccessGrant)
-	if err != nil {
-		return nil, err
-	}
-	project, err := uplink.OpenProject(ctx, access)
-	if err != nil {
-		return nil, err
-	}
-
-	return project, nil
-}
-
 func (o *ObjectStore) Init(config map[string]string) error {
 	o.log.Infof("objectStore.Init called")
 	if err := veleroplugin.ValidateObjectStoreConfigKeys(config, accessGrant); err != nil {
 		return err
 	}
-
-	project, err := setupUplink(context.Background(), config[accessGrant])
+	access, err := uplink.ParseAccess(config[accessGrant])
 	if err != nil {
 		return err
 	}
-	// defer project.Close()
-	o.project = project
+	o.access = access
 	return nil
 }
 
 func (o *ObjectStore) PutObject(bucket, key string, body io.Reader) error {
 	o.log.Infof("objectStore.PutObject called")
-	upload, err := o.project.UploadObject(context.Background(), bucket, key, nil)
+	project, err := uplink.OpenProject(context.Background(), o.access)
+	if err != nil {
+		return err
+	}
+	defer project.Close()
```

Comment:
I am not convinced about this change. The documentation of the `Init` method says that it "...returns an error if the ObjectStore cannot be initialized from the provided config.":

https://github.com/vmware-tanzu/velero/blob/f42c63af1b9af445e38f78a7256b1c48ef79c10e/pkg/plugin/velero/object_store.go#L27-L30

I think `OpenProject` is better in determining if the object store can be initialized than just parsing the access grant.

Moreover, opening and closing the project has its performance cost. If a velero command invokes many of these methods, it will be slower if every method opens and closes the project separately.

Here is such an example:

https://github.com/vmware-tanzu/velero/blob/f42c63af1b9af445e38f78a7256b1c48ef79c10e/pkg/persistence/object_store.go#L301-L311

Reply:
If you're worried about the performance issue, I guess suggesting that we open/close the project in the init method to check that everything is ok won't be a good idea to you, will it? :smiley: 

I'm not sure either about this, but it does feel weird to never close the project.